src/triage/component/catwalk/evaluation.py

```diff
@@ -1,24 +1,112 @@
 import functools
+import io
 import logging
 import time
+import warnings
 
 import numpy
+import pandas
 from sqlalchemy.orm import sessionmaker
 
 from . import metrics
-from .utils import db_retry, sort_predictions_and_labels
+from .utils import (
+    db_retry,
+    sort_predictions_and_labels,
+    get_subset_table_name,
+    filename_friendly_hash
+)
+
+
+def subset_labels_and_predictions(
+    subset_df,
+    labels,
+    predictions_proba,
+):
+    """Reduce the labels and predictions to only those relevant to the current
+       subset.
+
+    Args:
+        subset_df (pandas.DataFrame) A dataframe whose index is the entity_ids
+            and as_of_dates in a subset
+        labels (pandas.Series) A series of labels with entity_id and as_of_date
+            as the index
+        predictions_proba (numpy.array) An array of predictions for the same
+            entity_date pairs as the labels and in the same order
+
+    Returns: (pandas.Series, numpy.array) The labels and predictions that refer
+        to entity-date pairs in the subset
+    """
+    indexed_predictions = pandas.Series(predictions_proba, index=labels.index)
+    
+    # The subset isn't specific to the cohort, so inner join to the labels/predictions
+    labels_subset = labels.align(subset_df, join="inner")[0]
+    predictions_subset = indexed_predictions.align(subset_df, join="inner")[0].values
+
+    logging.debug(
+        "%s entities in subset out of %s in matrix.",
+        len(labels_subset),
+        len(labels),
+    )
+
+    return (labels_subset, predictions_subset)
+
+
+def query_subset_table(db_engine, as_of_dates, subset_table_name):
+    """Queries the subset table to find the entities active at the given
+       as_of_dates
+
+    Args:
+        db_engine (sqlalchemy.engine) a database engine
+        as_of_dates (list) the as_of_Dates to query
+        subset_table_name (str) the name of the table to query
+
+    Returns: (pandas.DataFrame) a dataframe indexed by the entity-date pairs
+        active in the subset
+    """
+    as_of_dates_sql = "[{}]".format(
+        ", ".join("'" + date.strftime("%Y-%m-%d %H:%M:%S.%f") + "'" for date in as_of_dates)
```

Comment:
would be nicer but no big deal:

```suggestion
        ", ".join("'{}'".format(date.strftime("%Y-%m-%d %H:%M:%S.%f")) for date in as_of_dates)
```